apps/acrs/behaviours/aggregator-save-update.js

```diff
@@ -77,14 +81,19 @@ module.exports = superclass => class extends superclass {
 
     const newItem = { itemTitle, fields };
 
-    if (aggregateLimit) {
-      if (items.length < aggregateLimit) {
+    if(isUpdate) {
+      items.splice(updatingIndex, 0, newItem);
+    }else{
+      if (aggregateLimit) {
+        if (items.length < aggregateLimit) {
+          items.push(newItem);
+        }
+      } else {
         items.push(newItem);
       }
-    } else {
-      items.push(newItem);
     }
 
+
     this.setAggregateArray(req, items);
     res.redirect(`${req.baseUrl}${req.form.options.route}`);
```

Comment:
You could make use of the existing session item `'aggregator-edit-id'` if you like - then splice every time and use the items array length as a backup `updatingIndex` for adding new items.

If you want to do this you would have to stop the various limit-xxxxxx.js behaviours from unsetting the `'aggregator-edit-id'` value.

If not make sure to reset your values for `isUpdate` and `updatingIndex`.


```
    const newItem = { itemTitle, fields };
    const updatingIndex = req.sessionModel.get('aggregator-edit-id') || items.length;

    if (aggregateLimit) {
      if (items.length < aggregateLimit) {
        items.splice(updatingIndex, 0, newItem);
      }
    } else {
      items.splice(updatingIndex, 0, newItem);
    }

    req.sessionModel.unset('aggregator-edit-id');
    this.setAggregateArray(req, items);
    res.redirect(`${req.baseUrl}${req.form.options.route}`);
```

Reply:
i have reset isUpdate and updatingIndex.